### PR TITLE
Migrate to the new SDK packages

### PR DESCRIPTION
**Pull request by smarlowucf**
Drop requirement on discovery API

### CODE REVIEW

**Review by apozsuse — APPROVED** [review]
LGTM :+1:

**Comment by apozsuse** [comment]
There's another import (for one exception) that has to be removed as it's causing the tests to fail.
https://github.com/SUSE-Enceladus/gceimgutils/blob/1175c97458af1a8afc17a431f34488239d8e639e/lib/gceimgutils/gceutils.py#L31

**Comment by smarlowucf** [comment]
> There's another import (for one exception) that has to be removed as it's causing the tests to fail.
> 
> https://github.com/SUSE-Enceladus/gceimgutils/blob/1175c97458af1a8afc17a431f34488239d8e639e/lib/gceimgutils/gceutils.py#L31

Yeah, I was trying to figure out what error the new API will throw. But left it generic for now.

**Comment by smarlowucf** [comment]
Okay, this is ready for review. But requires coordination for merge and release.

File: lib/gceimgutils/gceutils.py [hunk 1]
```diff
@@ -187,3 +189,53 @@ def _no_name_warning(image, log_callback):
     msg = 'WARNING: Found image with no name, ignoring for search results. '
     msg += 'Image ID: %s' % image['ImageId']
     log_callback.info(msg)
+
+
+# ----------------------------------------------------------------------------
+def image_to_dict(image):
+    return {
```
Comment (smarlowucf) [inline lib/gceimgutils/gceutils.py:196]: I left the keys camel case to match the existing API output. However, if we push a major release then we could use snake case and thus the dictionary could then be used to create an object of Image class.

**Review by apozsuse — APPROVED** [review]
LGTM :+1:

File: python3-gceimgutils.spec [hunk 2]
```diff
@@ -26,16 +26,32 @@ License:        GPL-3.0+
 Group:          System/Management
 Url:            https://github.com/SUSE-Enceladus/gceimgutils
 Source0:        %{upstream_name}-%{version}.tar.bz2
+%if 0%{?sle_version} >= 150400
```
Comment (rjschwei) [inline python3-gceimgutils.spec]: The file should be renamed to `python-gceimgutils` and let's setup the spec file like ec2imgutils https://github.com/SUSE-Enceladus/ec2imgutils/blob/master/python-ec2imgutils.spec

**Review by rjschwei — CHANGES_REQUESTED** [review]

File: python3-gceimgutils.spec (hunk 2 above)
Comment (smarlowucf) [reply python3-gceimgutils.spec]: Updated. Should be ready when 3.11 switchover happens.

Comment (smarlowucf) [reply python3-gceimgutils.spec]: Wasn't sure if the libalternatives are required here.

Comment (smarlowucf) [reply python3-gceimgutils.spec]: Spec was merged separately. Rebased this PR to pull in spec.

**Pull request merged**